src/map.rs

```diff
@@ -257,6 +267,84 @@ impl Map {
     }
 }
 
+/// Represents the way tiles are laid out in a map.
+#[derive(Debug, PartialEq, Eq, Copy, Clone, Default)]
+#[allow(missing_docs)]
+pub enum StaggerIndex {
+    #[default]
+    Even,
+    Odd,
+}
+
+#[derive(Debug)]
+/// An error arising from trying to parse an [`StaggerIndex`] that is not valid.
+pub struct StaggerIndexError {
+    /// The invalid string found.
+    pub str_found: String,
+}
+
+impl std::fmt::Display for StaggerIndexError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_fmt(format_args!(
+            "failed to parse orientation, valid options are `even`, `odd` \
+        but got `{}` instead",
+            self.str_found
+        ))
+    }
+}
+
+impl FromStr for StaggerIndex {
+    type Err = StaggerIndexError;
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+        match s {
+            "even" => Ok(StaggerIndex::Even),
+            "odd" => Ok(StaggerIndex::Odd),
+            _ => Err(StaggerIndexError {
+                str_found: s.to_owned(),
+            }),
+        }
+    }
+}
+
+/// Represents the way tiles are laid out in a map.
```

Comment:
Documentation copied from `Orientation` enum. For `StaggerAxis`, I would suggest:
```
// Specifies which axis is staggered. Only applies to Staggered and Hexagonal
// map orientations.
```